### PR TITLE
Enables Getting Dataset Ids

### PR DESCRIPTION
**Pull request by DanielMwendwa**

### CODE REVIEW

File: coda_v2_python_client/firebase_client_wrapper.py [hunk 1]
```diff
@@ -36,3 +36,61 @@ def init_client(cls, crypto_token_path, app_name="CodaV2Client"):
         cred = credentials.Certificate(crypto_token_path)
         app = firebase_admin.initialize_app(cred, name=app_name)
         return cls(firestore.client(app))
+
+    def get_dataset_ids():
+        """
+        Gets unique ids of datasets having included only one id for segmented dataset ids.
```
Comment (as2388) [inline coda_v2_python_client/firebase_client_wrapper.py:42]: I would try to write these in terms of datasets where you can, with almost a side-note about segments. This is so users don't need to worry about segments when interacting with datasets.

e.g. something like "Gets all the available dataset ids in Coda. For each segmented dataset, returns only the primary dataset id"

File: coda_v2_python_client/firebase_client_wrapper.py [hunk 2]
```diff
@@ -36,3 +36,61 @@ def init_client(cls, crypto_token_path, app_name="CodaV2Client"):
...
+            segment_count = get_segment_count(dataset_id)
+            if segment_count is not None and segment_count > 1:
+                for segment_index in range(2, segment_count + 1):
+                    dataset_ids.remove(id_for_segment(dataset_id, segment_index))
+
+        return dataset_ids
+
+    def get_segment_ids():
+        """
+        Gets ids of all datasets including segmented dataset ids.
```
Comment (as2388) [inline coda_v2_python_client/firebase_client_wrapper.py:61]:
```suggestion
        Gets ids of all segments (including for datasets that contain only one segment)
```

File: coda_v2_python_client/firebase_client_wrapper.py [hunk 3]
```diff
@@ -36,3 +36,61 @@ def init_client(cls, crypto_token_path, app_name="CodaV2Client"):
...
+        :param dataset_id: Id for a datasets including segemented datasets
+        :type dataset_id: str
+        :param segment_index: Index in the range of segment count starting from index two, defaults to None
+        :type segment_index: int, optional
+        :return: Output of concatinating dataset id & segment index
+        :rtype: str
+        """
+        if segment_index is None or segment_index == 1:
+            return dataset_id
+        return dataset_id + f'_{segment_index}'
```
Comment (as2388) [inline coda_v2_python_client/firebase_client_wrapper.py:84]:
```suggestion
        return f"{dataset_id}_{segment_index}"
```
Now is a good opportunity to upgrade to f-strings where you see them

File: coda_v2_python_client/firebase_client_wrapper.py [hunk 4]
```diff
@@ -36,3 +36,61 @@ def init_client(cls, crypto_token_path, app_name="CodaV2Client"):
...
+
+    def get_segment_ids():
+        """
+        Gets ids of all datasets including segmented dataset ids.
+
+        :return: Ids of all datasets including segmented dataset ids.
+        :rtype: list of str
+        """
+        ids = []
+        for segment in self._client.collection(u'datasets').get():
```
Comment (as2388) [inline coda_v2_python_client/firebase_client_wrapper.py:67]:
```suggestion
        for segment in self._client.collection("datasets").get():
```
we only need to support py3

File: coda_v2_python_client/firebase_client_wrapper.py [hunk 5]
```diff
@@ -36,3 +36,61 @@ def init_client(cls, crypto_token_path, app_name="CodaV2Client"):
...
+            ids.append(segment.id)
+        return ids
+
+    def id_for_segment(dataset_id, segment_index=None):
+        """
+        Generates id for segmented dataset ids by concatinating dataset id & segment index.
```
Comment (as2388) [inline coda_v2_python_client/firebase_client_wrapper.py:73]:
```suggestion
        Gets the id for segment `n` of a dataset.
```

**Review by as2388 — APPROVED** [review]
Thanks, this looks great. 

One general comment about the docstrings: try to say what the function does rather than how it does it e.g. "gets the id for the nth segment of a dataset" rather than "gets an id by concatenating the dataset id to the segment index".

Let me know if you need any advice on what to prioritise adding next.

File: coda_v2_python_client/firebase_client_wrapper.py (hunk 1 above)
Comment (DanielMwendwa) [reply coda_v2_python_client/firebase_client_wrapper.py:44]: Thanks updated

**Pull request merged**